src/create-theme-provider.js

```diff
@@ -0,0 +1,92 @@
+const React = require('react');
+const PropTypes = require('prop-types');
+const isFunction = require('is-function');
+const isPlainObject = require('is-plain-object');
+const createBroadcast = require('./create-broadcast');
+const channel = require('./channel');
+
+/**
+ * Provide a theme to an entire react component tree via context
+ * and event listeners (have to do both context
+ * and event emitter as pure components block context updates)
+ */
+
+function createThemeProvider(CHANNEL = channel) {
+  return class ThemeProvider extends React.Component {
+    static propTypes = {
+      children: PropTypes.element,
+      theme: PropTypes.oneOfType([PropTypes.shape({}), PropTypes.func])
+        .isRequired,
+    };
+
+    static contextTypes = {
+      [CHANNEL]: PropTypes.func,
+    };
+
+    static childContextTypes = {
+      [CHANNEL]: PropTypes.func.isRequired,
+    };
+
+    constructor() {
+      super();
+      this.getTheme = this.getTheme.bind(this);
+    }
+
+    getChildContext() {
+      return { ...this.context, [CHANNEL]: this.broadcast.subscribe };
```

Comment:
Shouldn't [channel] export both: subscribe and unsubscribe? Are you sure users will only need subscribe?